ChromiumHtmlToPdfLib/Connection.cs

```diff
@@ -119,27 +114,30 @@ internal Connection(string url, int timeout, Logger logger)
         _webSocket = new ClientWebSocket();
         _receiveLoopCts = new CancellationTokenSource();
         OpenWebSocketAsync().GetAwaiter().GetResult();
-        Task.Factory.StartNew(ReceiveLoop, _receiveLoopCts.Token, TaskCreationOptions.LongRunning, TaskScheduler.Default);
+        Task.Factory.StartNew(ReceiveLoop, new ReceiveLoopState(_logger, _webSocket, OnMessageReceived, _receiveLoopCts.Token), _receiveLoopCts.Token, TaskCreationOptions.LongRunning, TaskScheduler.Default);
```

Comment:
refactored a bit to remove thread use of instance fields directly. simplifies nullability analysis and helps to avoid concurrency issues with mutable fields